File: veterinaria-system/frontend/src/models/Rol.js
/**
 * Modelo Rol
 * Arquitectura MVC - Capa de Modelos
 */

class Rol {
  constructor(data = {}) {
    this.id = data.id || null;
    this.nombre = data.nombre || '';
    this.descripcion = data.descripcion || '';
    this.created_at = data.created_at || null;
  }

  static fromJSON(json) {
    return new Rol(json);
  }

  toJSON() {
    return {
      id: this.id,
      nombre: this.nombre,
      descripcion: this.descripcion
    };
  }
}

export default Rol;

File: veterinaria-system/backend/api/models/rol.py
"""
Modelo Rol - Arquitectura MVC
"""
from django.db import models


class Rol(models.Model):
    """
    Modelo para definir los roles de usuario en el sistema.
    """
    ROLES_CHOICES = [
        ('administrador', 'Administrador'),
        ('veterinario', 'Veterinario'),
        ('tutor', 'Tutor'),
    ]

    nombre = models.CharField(max_length=50, unique=True, choices=ROLES_CHOICES)
    descripcion = models.TextField(blank=True)
    created_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        verbose_name = 'Rol'
        verbose_name_plural = 'Roles'
        ordering = ['nombre']

    def __str__(self):
        return self.nombre
